Package.swift

```diff
@@ -27,6 +27,7 @@ let cxxSettings: [CXXSetting] = [
     .define("REALM_VERSION_PATCH", to: String(coreVersionExtra[0])),
     .define("REALM_VERSION_EXTRA", to: "\"\(coreVersionExtra.count > 1 ? String(coreVersionExtra[1]) : "")\""),
     .define("REALM_VERSION_STRING", to: "\"\(coreVersionStr)\""),
+    .define("REALM_ASYNC_WRITES", .when(configuration: .debug)),
```

Comment:
Why isn't this just in the test setting for now?

Reply:
Only to reduce the availability of a feature before making it fully accessible.

Reply:
I worry that this may end up confusing for users, as this will expose it to them until they go to release their app. The idea of feature flagging was that users would have to explicitly opt in.

Reply:
This is required as a value I add for tests will take effect on tests build only. The framework build will have no async code.

Reply:
When Realm is installed via SPM, debug builds of the app use debug builds of Realm, and this will be set.

Reply:
Actually I think this might be fine? It'll be defined when building Realm itself but not for the user's code which imports RLMRealm.h, so while the methods are defined the user won't be able to see them.